### PR TITLE
Engine fixes

### PR DESCRIPTION
**Pull request by DreierF**
- [ ] [No new Teamscale findings](https://build.cqse.eu/teamscale/delta.html#input/jacoco-client/?showMergeFindings=true)
- [ ] [Changes are tested adequately](https://build.cqse.eu/teamscale/tests.html#/jacoco-client/)
- [ ] README.md updated in case of user-visible changes
- [ ] CHANGELOG.md updated
- [ ] build.gradle version number increased according to semantic versioning

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -12,7 +12,7 @@ buildscript {
 
 ext {
 	appVersion = '11.3.1' // Also update version in BuildVersion.kt
-	pluginVersion = '0.4.1' // Also update version in BuildVersion.kt
+	pluginVersion = '0.4.2' // Also update version in BuildVersion.kt
```
Comment (albertsteckermeier) [inline build.gradle]: For Teamscale we have 0.4.6 set as the version. Either bump version here or CR#0 it on the Teamscale master.

Comment (DreierF) [reply build.gradle]: Oh I completely forgot to push these changes. Version was already set to 0.4.6.

File: teamscale-client/src/main/java/com/teamscale/client/FileLoggingInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	/** Constructor. */
+	public FileLoggingInterceptor(File file) {
+		this.file = file;
+	}
+
+	@Override
+	public Response intercept(Chain chain) throws IOException {
+		Request request = chain.request();
+
+		long t1 = System.nanoTime();
```
Comment (albertsteckermeier) [inline teamscale-client/src/main/java/com/teamscale/client/FileLoggingInterceptor.java]: t1 --> requestStartTime?

File: teamscale-client/src/main/java/com/teamscale/client/FileLoggingInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+			fileWriter.write(requestBuffer.readUtf8());
+
+			Response response;
+			try {
+				response = chain.proceed(request);
+			} catch (Exception e) {
+				fileWriter.write("\n\nRequest failed!\n");
+				e.printStackTrace(fileWriter);
+				throw e;
+			}
```
Comment (albertsteckermeier) [inline teamscale-client/src/main/java/com/teamscale/client/FileLoggingInterceptor.java]: Extract method getResponse

File: teamscale-client/src/main/java/com/teamscale/client/FileLoggingInterceptor.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+			long t2 = System.nanoTime();
```
Comment (albertsteckermeier) [inline teamscale-client/src/main/java/com/teamscale/client/FileLoggingInterceptor.java]: t2 --> requestEndTime

File: teamscale-client/src/main/java/com/teamscale/client/TeamscaleClient.java [hunk 5]
```diff
@@ -37,6 +45,25 @@ public TeamscaleClient(String baseUrl, String user, String accessToken, String p
 	public Response<List<PrioritizableTestCluster>> getImpactedTests(List<ClusteredTestDetails> testList, Long baseline,
 																	 CommitDescriptor endCommit,
 																	 String partition) throws IOException {
+		long startTime = System.currentTimeMillis();
+		Response<List<PrioritizableTestCluster>> impactedTestsResponse;
+		do {
+			impactedTestsResponse = getImpactedTestsOnce(testList, baseline,
+					endCommit, partition);
+			// Retry for up to one minute until the coverage uploads are processed by Teamscale
+		} while (impactedTestsResponse.code() == 412 && System.currentTimeMillis() - startTime < 60000);
```
Comment (albertsteckermeier) [inline teamscale-client/src/main/java/com/teamscale/client/TeamscaleClient.java]: This method never returns.

File: teamscale-client/src/main/java/com/teamscale/client/FileLoggingInterceptor.java [hunk 6]
```diff
@@ -66,4 +59,16 @@ public Response intercept(Chain chain) throws IOException {
 			return response.newBuilder().body(wrappedBody).build();
 		}
 	}
+
+	private Response getResponse(Chain chain, Request request, PrintWriter fileWriter) throws IOException {
+		Response response;
+		try {
+			response = chain.proceed(request);
```
Comment (albertsteckermeier) [inline teamscale-client/src/main/java/com/teamscale/client/FileLoggingInterceptor.java]: Return here to avoid variable.

**Pull request merged**